### PR TITLE
error: use int64_t as reference counter

### LINKED ISSUE #4902: Make struct error reference counter uint64/int64
After https://github.com/tarantool/tarantool/commit/ca79e1cfba90cc2cb7ab9de229374366525c0749 it becomes unsafe to use `error.prev`, `box.error.last()`, or strictly speaking any other error method working with an existing error object. Any of them now can throw a Lua error. That should be fixed by making the reference counter uint64/int64. The issue may seem artificial - after all it is not that simple to overflow uint32 counter. But as #1166 and #3224 show, it is real. So rather than complicate user's already not so easy life better make the counter uint64. This is not a memory critical object so as to bother so much about making it as small as possible.

### PR DESCRIPTION
**Pull request by dmitryikh**
Before it was unsafe to use error.prev, box.error.last(), or any error
method that increment error reference counter. Any of them can throw
a Lua error in case of INT32_MAX overflow.

This patch uses int64_t as error reference counter making it impossible
to get counter overflow.

Closes #4902

Unfortunately I don't know how to write tests on this behaviour..

### CODE REVIEW

**Comment by dmitryikh** [comment]
Should I send the PR to mailing list?

**Comment by Gerold103** [comment]
I will take a look here.

**Comment by Gerold103** [comment]
> Before it was unsafe to use error.prev, box.error.last(), or any error
method that increment error reference counter. Any of them can throw
a Lua error in case of INT32_MAX overflow.
>
>This patch uses int64_t as error reference counter making it impossible
to get counter overflow.
>
>Closes #4902

Typo: 'increment' -> 'increments'. Also in the commit message we describe the old behaviour using past tense. You said "any of them can throw", but you should say "any of them could throw".

File: src/lua/error.lua [hunk 1]
```diff
@@ -39,6 +39,9 @@ error_set_prev(struct error *e, struct error *prev);
 const char *
 box_error_custom_type(const struct error *e);
 
+void
+error_ref(struct error *e);
```
Comment (Gerold103) [inline src/lua/error.lua:43]: To be able to use `error_ref` in Lua via FFI you should export `error_ref` symbol. Add it to `exports.h` file, respecting the alphabetic order of symbols there.

**Comment by dmitryikh** [comment]
Thanks, @Gerold103, for review!

I fixed the issues.

**Comment by Gerold103** [comment]
You may need a second review, and someone who will merge. @kyukhin probably.

**Comment by kyukhin** [comment]
Could you please send your final patch to tarantool-patches mailing list?

**Comment by dmitryikh** [comment]
> Could you please send your final patch to tarantool-patches mailing list?

done

**Comment by Totktonada** [comment]
It was pushed as [2.5.0-129-g2e3c81dee](https://github.com/tarantool/tarantool/commit/2e3c81deea505cb5a03d606de7373bb41b559cd9) to `master` and as [2.4.1-63-g56b57be53](https://github.com/tarantool/tarantool/commit/56b57be538302969f2f3f9dea12aeae6346ee2aa) to `2.4`.

@kyukhin I updated drafts of 2.5.1 and 2.4.2 release notes.

**Pull request closed without merging**